packages/lodestar/src/chain/stateTransition/util/attestation.ts

```diff
@@ -150,3 +150,33 @@ export function isValidAttestationSlot(
     currentSlot <= attestationSlot + config.params.SLOTS_PER_EPOCH
   );
 }
+
+/**
+ * Aggregates two attestation.
+ * @param config
+ * @param attestation1
+ * @param attestation2
+ */
+export function aggregateAttestation(
+  config: IBeaconConfig,
+  attestation1: Attestation,
+  attestation2: Attestation
+): Attestation {
+  const aggregatedAttestation: Attestation = clone(attestation2, config.types.Attestation);
+  let hasMoreSignatures = false;
+  for(let i = 0; i < attestation1.aggregationBits.bitLength; i++) {
+    if(attestation1.aggregationBits.getBit(i) !== attestation2.aggregationBits.getBit(i)) {
+      aggregatedAttestation.aggregationBits.setBit(i, true);
+      hasMoreSignatures = true;
+    }
+  }
+  for(let i = 0; i < attestation1.custodyBits.bitLength; i++) {
+    if(attestation1.custodyBits.getBit(i)) {
+      aggregatedAttestation.custodyBits.setBit(i, true);
+    }
+  }
+  if(hasMoreSignatures) {
+    aggregatedAttestation.signature = aggregateSignatures([attestation1.signature, attestation2.signature]);
```

Comment:
do we need to make sure these 2 attestations have same data?

Reply:
Yeah, but it would slow down performance and caller is always checking anyways. But this method needs more checking as this doesn't work like this exactly

Reply:
I will probably add that check later on

Reply:
Is there any room to improve the performance? If `AttestationData` is immutable we can possibly cache its `hashTreeRoot` value inside the object.

Reply:
I think there is somewhere issue to hache common hashTreeRoots inside ssz itself